skyway_app/static/map.js

```diff
@@ -25,16 +35,21 @@ function askForUserLocation(){
 }
 
 function addListeners(){
+
   // On Click, remove all pins, add a pin, save location
   map.addListener('click', function(event) {
-    //TODO remove all pins
+    //clear any existing marker
+    for (var i = 0; i<markers.length;i++){
```

Comment:
Won't work. Suppose it's because you are deleting a pointer but not the marker itself?